csharp/src/Apache.Arrow/C/CArrowArrayImporter.cs

```diff
@@ -90,42 +96,49 @@ public static unsafe RecordBatch ImportRecordBatch(CArrowArray* ptr, Schema sche
 
         private sealed unsafe class ImportedArrowArray : ImportedAllocationOwner
         {
-            private readonly CArrowArray* _cArray;
+            private readonly CArrowArray _cArray;
 
             public ImportedArrowArray(CArrowArray* cArray)
             {
                 if (cArray == null)
                 {
                     throw new ArgumentNullException(nameof(cArray));
                 }
-                _cArray = cArray;
-                if (_cArray->release == null)
+                if (cArray->release == null)
                 {
                     throw new ArgumentException("Tried to import an array that has already been released.", nameof(cArray));
                 }
+                _cArray = *cArray;
+                cArray->release = null;
             }
 
             protected override void FinalRelease()
             {
-                if (_cArray->release != null)
+                if (_cArray.release != null)
                 {
-                    _cArray->release(_cArray);
+                    fixed (CArrowArray* cArray = &_cArray)
+                    {
+                        cArray->release(cArray);
+                    }
```

Comment:
Without the "fixed", in theory the GC could move the current object while we're inside the call. This would change the location of the embedded struct.

Reply:
Yes, I was wondering about the reason. Thank you!